roles/cfg_openwrt/templates/common/config/freifunk.j2

```diff
@@ -2,8 +2,18 @@ config public 'contact'
 {% if contact_name is defined %}
 	option name '{{ contact_name }}'
 {% endif %}
+{% if nickname is defined %}
 	option nickname '{{ contact_nickname }}'
-	option mail '{{ contact_email }}'
+{% endif %}
+{% if contact_email is defined %}
```

Comment:
This should be implemented differently so it is backwards compatible:
If contact_email is set it should simply be used as an entry in contacts. If the site has the community flag and contacts is empty the default should be used. If it is absent AND the community flag is false it should fail.

Reply:
> If contact_email is set it should simply be used as an entry in contacts.

We could do it that way. But in my opinion, allowing (legacy) email-contacts-field might introduce unnecessary complexity. My intention was to encourage each site maintainer to migrate their code to the new standard by their self, once they build the site next time. That way, we can have migrated the whole repo in a year or so.

> If the site has the community flag and contacts is empty the default should be used.

In the mentioned [wednesday meeting](https://lists.berlin.freifunk.net/pipermail/berlin/2022-September/053153.html) we agreed on plus-addressing. It has some advantages, like we can sort the incoming emails better. All Emails addresses to info+SOMETHING@berlin will go to the regular info-address. So info+SITE@berlin should be the new standard. There's no change in the current receiving infrastructure.


> If it [ contacts-field ] is absent AND the community flag is false it should fail.

That's already happening, isn't it? Do you think, we should require the community flag to be set explicitly to false, instead of leaving it undefined?

Reply:
> > If contact_email is set it should simply be used as an entry in contacts.
> 
> We could do it that way. But in my opinion, allowing (legacy) email-contacts-field might introduce unnecessary complexity. My intention was to encourage each site maintainer to migrate their code to the new standard by their self, once they build the site next time. That way, we can have migrated the whole repo in a year or so. Breaking backward compability was by intention here.

Ok, got the idea. But in that case please migrate the contact_email fields from all existing locations to the contacts list together with the pull request and add the community flag for the community locations. From what I can see (`grep -r "contact_email" .` within `group_vars`) most configurations are just fine and forcing all those people to migrate instead of just migrating for them is bad style.

> 
> > If the site has the community flag and contacts is empty the default should be used.
> 
> In the mentioned [wednesday meeting](https://lists.berlin.freifunk.net/pipermail/berlin/2022-September/053153.html) we agreed on plus-addressing. It has some advantages, like we can sort the incoming emails better. All Emails addresses to info+SOMETHING@berlin will go to the regular info-address. So info+SITE@berlin should be the new standard. There's no change in the current receiving infrastructure.

With this I ment the proposed new default (info+SITE@berlin).

> 
> > If it [ contacts-field ] is absent AND the community flag is false it should fail.
> 
> That's already happening, isn't it? Do you think, we should require the community flag to be set explicitly to false, instead of leaving it undefined?

The idea here was that a site must be a community site (contact set or contact mail generated as proposed) or if not setting a contact field is mandatory.

In general I think having the community flag is a really good idea as it would allow us to differentiate sites that can be mass flashed by the community vs. sites that are operated by a single person and can not or should not be mass flashed. I think the community flag should be set to true only for the community sites and should otherwise be absent.